### PR TITLE
Add Microsoft distribution of the OpenTelemetry starter

### PR DESCRIPTION
**Pull request by jeanbisutti**

### CODE REVIEW

File: data/ecosystem/distributions.yaml [hunk 1]
```diff
@@ -75,6 +75,10 @@
   url: https://github.com/microsoft/ApplicationInsights-Java
   docsUrl: https://learn.microsoft.com/azure/azure-monitor/app/opentelemetry-enable?tabs=java
   components: [Java]
+- name: Azure Monitor OpenTelemetry Distro / Application Insights in Spring Boot native image Java application
```
Comment (theletterf) [inline data/ecosystem/distributions.yaml:78]:
```suggestion
- name: Azure Monitor OpenTelemetry Distro
```

Any chance we could shorten it to the above?

Comment (jeanbisutti) [reply data/ecosystem/distributions.yaml]: For the Microsoft distribution of the OTel Java agent, the name is `Azure Monitor Distribution of OpenTelemetry Java`.

What do you think about `Azure Monitor Distribution of OpenTelemetry Java Spring Boot starter`, or `Azure Monitor Distribution of OpenTelemetry Java starter`?

Other vendors providing a distribution of the OpenTelemetry Spring Boot starter could follow the same pattern.

Comment (theletterf) [reply data/ecosystem/distributions.yaml]: Suggesting the following pattern:

- Azure Monitor Distribution of OpenTelemetry Java (Spring Boot starter)

Comment (jeanbisutti) [reply data/ecosystem/distributions.yaml]: Great! I have updated the PR with this formulation.

**Comment by jeanbisutti** [comment]
I have rebased and fixed the merge conflict.

**Review by cartermp — APPROVED** [review]

**Pull request merged**